tests/utils.py

```diff
@@ -1,15 +1,16 @@
 from typing import Optional
 
 from fastapi import Response
-from httpx import AsyncClient
+from httpx import AsyncClient, Cookies
 
 from app import models, repository
-from app.main import get_app
-from app.oauth2 import create_access_token
+from app.auth_manager import get_strategy
+from app.config import settings
+from app.main import app
 from app.utils.enums import RequestMethod
 
 
-def authorized_httpx_client(client: AsyncClient, user: models.User):
+async def authorized_httpx_client(client: AsyncClient, user: models.User):
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (testing):** Ensure `authorized_httpx_client` handles user not found scenarios.

It would be beneficial to include a test or a check within this utility function to handle cases where the user does not exist, ensuring the tests remain robust under various conditions.

```suggestion
async def authorized_httpx_client(client: AsyncClient, user: Optional[models.User]):
    """
    Fixture that provides an authorized client with a token.

    Returns:
        AsyncClient: An authorized client with the provided token.
    """
    if user is None:
        raise ValueError("User does not exist.")

    strategy = get_strategy()
    token = strategy.write_token(user)

    client.cookies = {
        **client.cookies,
        settings.access_token_name: await token,
    }
    return client
```